pulpcore/app/viewsets/user.py

```diff
@@ -164,10 +173,33 @@ class GroupObjectPermissionViewSet(NamedModelViewSet):
     serializer_class = PermissionSerializer
     queryset = Permission.objects.all()
 
+    def get_object_pk(self, request):
+        """Get object pk."""
+
+        if "obj" not in request.data:
+            raise ValidationError(_("Please provide 'obj' value"))
+        try:
+            obj_pk = request.data["obj"].strip("/").split("/")[-1]
+            uuid.UUID(obj_pk)
+        except (AttributeError, ValueError):
+            raise ValidationError(_(f"Invalid value for 'obj': {request.data['obj']}"))
+
+        return obj_pk
+
     def get_model_permission(self, request):
         """Get model permission"""
+
+        if "permission" not in request.data:
+            raise ValidationError(_("Please provide 'permission' value"))
+
         codename = request.data["permission"].split(".")[-1]
-        return get_object_or_404(Permission, codename=codename)
+
+        try:
+            permission = Permission.objects.get(codename=codename)
+        except (Permission.MultipleObjectsReturned, Permission.DoesNotExist, FieldError) as exc:
+            raise ValidationError(_(str(exc)))
```

Comment:
```suggestion
            raise ValidationError(str(exc))
```
Won't work at all, but the exception should only use translatable strings anyway.